### PR TITLE
Bug/109 bug scale indicators are not linked to waves

### PR DESCRIPTION
**Pull request by leandertolksdorf**

### CODE REVIEW

**Review by jenswaechtler — CHANGES_REQUESTED** [review]
image

- [x] Zero Indicator should be clearly visible in light theme
- [x] Zero Indicator should move with the other indicators. 

Other problem with the linking to the waves: 
- [ ] Ball indicator still moves if wave is shut off 
- [ ] Line indicators still move if wave is shut off

**Review by jenswaechtler — APPROVED** [review]
Now it works! Good job!

File: Apps/frontend/src/views/Indicators.svelte [hunk 1]
```diff
@@ -7,35 +7,61 @@
...
   let canvasElement;
   let canvasContext;
-  let min = Array(10).fill(0.0);
-  let max = Array(10).fill(0.0);
+  let current = Array(NUM_CHANNELS).fill(0.0);
+  let min = Array(NUM_CHANNELS).fill(0.0);
+  let max = Array(NUM_CHANNELS).fill(0.0);
+  let offsets = Array(NUM_CHANNELS).fill(0.0);
+  let scalings = Array(NUM_CHANNELS).fill(1.0);
+  let startStopLine = Array(NUM_CHANNELS).fill(true);
```
Comment (PhlppKrmr) [inline Apps/frontend/src/views/Indicators.svelte:22]: Wasn't quite sure what you ment with this variable, guess it is more like `isChannelRunning` or `isChannelStarted`/`hasChannelStarted`?

**Review by PhlppKrmr — CHANGES_REQUESTED** [review]
Apart from my comment, looks very good, only tests are missing to fulfill DoD.

One more thing to consider: Mr. Eckert told us to implement some sort of "bracket" from 0 to the max value (no need to indicate min). I don't know if this is still scope of this issue, but I think we don't actually need your bouncing balls and simply need a (more or less) static bracket (see [Issue 53](https://github.com/amosproj/amos2022ws03-software-oscilloscope/issues/53#issuecomment-1316952469)).

File: Apps/frontend/src/views/Indicators.svelte (hunk 1 above)
Comment (leandertolksdorf) [reply Apps/frontend/src/views/Indicators.svelte:22]: Good point, I'm not happy with the naming, too. But I stole it from the other components to have a consistent naming and change it in the next refactoring :D

Comment (leandertolksdorf) [reply Apps/frontend/src/views/Indicators.svelte:22]: Regarding the bracket, I don't think this is in scope of the issue.
But I think that would be a question for monday.

Comment (leandertolksdorf) [reply Apps/frontend/src/views/Indicators.svelte:22]: I added a snapshot testing module to cypress and one test testing the initial state of the indicators, however:
- I'm getting an error in cypress.config.js when importing the plugin without the .js extension, but without the extension cypress throws an error. Don't know why...
- For testing the functionality of offsets and scalings we need some way to mock input on the controls. For that we either need to mock the generator to send a static signal and simulate the controls with cy.click(), which would make the tests dependent on the generator. Or we control the indicators programmatically (better), but this is not possible with cypress from what I learned...

What's your opinion on that?

Comment (PhlppKrmr) [reply Apps/frontend/src/views/Indicators.svelte:22]: Sounds like a sticky situation. Sadly I'm not all that good with Cypress, maybe @jenswaechtler knows how to best approach this issue.

**Review by nicolaskolbenschlag — APPROVED** [review]

**Review by PhlppKrmr — CHANGES_REQUESTED** [review]
Currently your indicators don't update when the signal isn't running (turn on button), how hard do you think this is to change?

**Comment by leandertolksdorf** [comment]
> Currently your indicators don't update when the signal isn't running (turn on button), how hard do you think this is to change?

@PhlppKrmr 
That's on purpose, see https://github.com/amosproj/amos2022ws03-software-oscilloscope/pull/111#pullrequestreview-1202276163

**Review by motschel123 — APPROVED** [review]
LGTM.
Maybe we should signal somehow if an indicator is out side the visible scope.

**Comment by leandertolksdorf** [comment]
> > > Currently your indicators don't update when the signal isn't running (turn on button), how hard do you think this is to change?
> > 
> > 
> > @PhlppKrmr That's on purpose, see [#111 (review)](https://github.com/amosproj/amos2022ws03-software-oscilloscope/pull/111#pullrequestreview-1202276163)
> > Edit: Oh, you mean after mounting? Fixed that.
> 
> No, I mean this: image
> 
> When you start the signal (so we see something (Turn on button)), then stop it and then move the offset/amplitude slider, we don't update the indicators.

@PhlppKrmr got it, fixed

**Review by PhlppKrmr — APPROVED** [review]

**Pull request merged**